Admin/Extension/LockExtension.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin\Extension;
+
+use Sonata\AdminBundle\Admin\AdminExtension;
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Model\LockInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Emmanuel Vella <vella.emmanuel@gmail.com>
+ */
+class LockExtension extends AdminExtension
+{
+    protected $version;
```

Comment:
We can't map the ``_lock_version`` field to the object, because doctrine reads the version number directly from the object (not from the database) and then compare it to the version number we use when calling ``lock()``: https://github.com/doctrine/doctrine2/blob/v2.5.0/lib/Doctrine/ORM/UnitOfWork.php#L2326